Cargo.toml

```
@@ -101,13 +101,16 @@ windows = { version = "0.56.0", features = [
 ] }
 windows-registry = "0.2.0"
 
-[target.'cfg(not(target_os = "windows"))'.dependencies]
-skia-safe = { version = "0.75.0", features = ["gl", "textlayout"] }
-
 [target.'cfg(target_os = "macos")'.dependencies]
 objc2 = "0.5.2"
 objc2-foundation = { version = "0.2.2", features = [ "NSUserDefaults" ] }
-objc2-app-kit = { version = "0.2.2", features = [ "NSLayoutConstraint" ] }
+objc2-app-kit = { version = "0.2.2", features = [ "NSLayoutConstraint", "objc2-quartz-core" ] }
+objc2-quartz-core = { version = "0.2.2", features = [ "objc2-metal", "CALayer", "CAMetalLayer" ] }
+objc2-metal = { version = "0.2.2", features = [ "MTLCommandQueue", "MTLCommandBuffer" ] }
+skia-safe = { version = "0.75.0", features = ["metal", "gl", "textlayout"] }
```

```suggestion
skia-safe = { git = "https://github.com/rust-skia/rust-skia.git", branch = "master", features = ["metal", "gl", "textlayout"] }
```

we can assume to target upstream and bump the `0.79.0` afterwards when available which is coming very soon, we are free on code changes from the client side which has no breaking changes and only improvements made by skia-safe team.

we can also setup our CI correctly with no need of python version regression to make it work.

---

I'm afraid of whether it is proper to change the version to use git. This is much unstable and would lead to some building problem

---

And also probably a break of function suddenly without clear information on this.

---

I mean in the future, not in current version.

---

Yes, we should not use the git version, only the official versions have binaries available, and that's what 99% of the people are using when building Neovide. 

However, we should remove the opengl support on macOS, or request for that combination from skia-safe, if we still want to support it. There are no binaries for that combination available here https://github.com/rust-skia/skia-binaries/releases. That will fix the build problems for @falcucci, I think, but if I remember correctly @crupest has proxy problems that prevents the usage of the pre-built binaries.

---

> but if I remember correctly @crupest has proxy problems that prevents the usage of the pre-built binaries.

Actually, no. The proxy problem is I failed to access Discord. And with a proper proxy to make network work, everything works as intended. Prebuilt binary for Windows will be downloaded directly without being built from scratch. The problem is still there is no OpenGL+Metal prebuilt binary.

> However, we should remove the opengl support on macOS, or request for that combination from skia-safe, if we still want to support it.

I would choose to request for this combination from skia-safe. Or at least, we can get the reason why this combination is missing if it is intentional.

---

One thing we could do is to make metal a feature flag instead of command line parameter, and release two versions of Neovide. That way we never need to build the unsupported configurations.

---

I wasn't clear with my request I guess.

*we are not merging using the git target version* 

this change is to take advantage of upstream changes and bump the release `0.79.0` when is available, so we can setup our CI correctly installing ninja with less efforts and tricks.

The change I would still request and @fredizzimo is right is about change the branch from master but add a pinned to a specific commit (the current latest).

Regarding the OpenGL deprecation on neovide, I think we should still offer both, in the end of the day the final user will install the .dmg or just the binary in their PATH and in case they need to build it from source is just a matter of install ninja and that's it.

---

> One thing we could do is to make metal a feature flag instead of command line parameter, and release two versions of Neovide. That way we never need to build the unsupported configurations.

@fredizzimo This is good for us, developers, but causing a lot of trouble for users and delivering.

The users have to download two versions to test and then choose one. If they want to switch temporarily because of some bugs or other things, it is more inconvenient.

For delivering, like package manager homebrew, they have to deliver 2 versions with different names. It might confuse users a lot.

So I would rather to choose building skia and not use feature.

---

@crupest I agree about building skia, deliver two apps unfortunately is out of our intention so far.

I would take building skia as our prior solution but also *make a request to support both features to improve compilation time* but this is not a blocker for us.

---

we can still deprecate OpenGL for macOS in another future version, and then use their binaries.

---

> I would take building skia as our prior solution but also make a request to support both features to improve compilation time but this is not a blocker for us.

@falcucci Agree with you.

But one thing to notice is that current CI failure is not about ninja. It is python version problem. But we can use an GitHub Action to set python version before building.

---

the CI is currently failing because you are still using the `0.75.0` and not the upstream `0.79.0` and we need to install ninja anyway afterwards, there is no need to manage python version since it gets the latest.

steps to fix are probably:
- update skia to upstream using the last pinned commit;
- add ninja installation to our ci pipelines;

---

the CI is currently failing because you are still using the `0.75.0` and not the upstream `0.79.0` and we need to install ninja anyway afterwards, there is no need to manage python version since it gets the latest.

steps to fix are probably:
- update skia to upstream using the last pinned commit;
- add ninja installation to our ci pipelines;

---

I don't think we should merge anything that requires building from source, since that's a big complication for macOS contributor. We have four options.
1. Remove the opengl support
2. Keep the opengl support, but make it a feature flag, and add both variations to the CI and the release
3. Request for a new binary that includes both opengl and meta
4. Host our own skia binaries somewhere else.

---

I actually have no problem to remove OpenGL support. Metal is much more stable for macOS.

---

> big complication for macOS contributor.

I totally agree, that's a point I forgot. based on that if we don't have both features mashup from the skia team, I would stick to remove OpenGL support.